GDJS/GDJS/Events/CodeGeneration/MetadataDeclarationHelper.cpp

```diff
@@ -0,0 +1,1467 @@
+
+  objectMetadata
+      .AddScopedCondition("FlippedY", _("Vertically flipped"),
+                          _("Check if the object is vertically flipped"),
+                          _("_PARAM0_ is vertically flipped"), _("Effects"),
+                          "res/actions/flipY24.png", "res/actions/flipY.png")
+      .AddParameter("object", _("Object"), objectType)
+      .SetFunctionName("isFlippedY");
+
+  // Deprecated
+  objectMetadata
+      .AddCondition("FlippedY", _("Vertically flipped"),
+                    _("Check if the object is vertically flipped"),
+                    _("_PARAM0_ is vertically flipped"), _("Effects"),
+                    "res/actions/flipY24.png", "res/actions/flipY.png")
+      .SetHidden()
+      .AddParameter("object", _("Object"), objectType)
+      .SetFunctionName("isFlippedY");
+
+  objectMetadata
+      .AddExpressionAndConditionAndAction(
+          "number", "Opacity", _("Opacity"),
+          _("the opacity of an object, between 0 (fully transparent) to 255 "
+            "(opaque)"),
+          _("the opacity"), _("Visibility"), "res/conditions/opacity24.png")
+      .AddParameter("object", _("Object"), objectType)
+      .UseStandardParameters("number", gd::ParameterOptions::MakeNewOptions())
+      .SetFunctionName("setOpacity")
+      .SetGetter("getOpacity");
+
+  return objectMetadata;
+}
+
+/**
+ * Check if the name of the function is the name of a lifecycle function (for
+ * events-based behaviors), that will be called automatically by the game
+ * engine.
+ */
+bool MetadataDeclarationHelper::IsBehaviorLifecycleEventsFunction(
+    const gd::String &functionName) {
+  return functionName == "onCreated" || functionName == "onActivate" ||
+         functionName == "onDeActivate" || functionName == "doStepPreEvents" ||
+         functionName == "doStepPostEvents" || functionName == "onDestroy" ||
+         // Compatibility with GD <= 5.0 beta 75
+         functionName == "onOwnerRemovedFromScene";
+  // end of compatibility code
+}
+
+/**
+ * Check if the name of the function is the name of a lifecycle function (for
+ * events-based objects), that will be called automatically by the game engine.
+ */
+bool MetadataDeclarationHelper::IsObjectLifecycleEventsFunction(
+    const gd::String &functionName) {
+  return functionName == "onCreated" || functionName == "doStepPostEvents" ||
+         functionName == "onDestroy" || functionName == "onHotReloading";
+}
+
+/**
+ * Check if the name of the function is the name of a lifecycle function (for
+ * events-based extensions), that will be called automatically by the game
+ * engine.
+ */
+bool MetadataDeclarationHelper::IsExtensionLifecycleEventsFunction(
+    const gd::String &functionName) {
+  return gd::EventsFunctionsExtension::IsExtensionLifecycleEventsFunction(
+      functionName);
+}
+
+gd::String
+MetadataDeclarationHelper::RemoveTrailingDot(const gd::String &description) {
+  return description[description.length() - 1] == '.'
```

Comment:
`if (description.length() === 0) return ""` - otherwise it will crash

Reply:
I added a test.